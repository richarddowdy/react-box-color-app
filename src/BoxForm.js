import React, { useState } from 'react';

function BoxForm({ addBox }) {

  const INITIAL_STATE = { height: 0, width: 0, color: "" };
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleSubmit = evt => {
    evt.preventDefault();
    addBox(formData);
    setFormData(INITIAL_STATE);
  };

  const handleChange = evt => {
    const { name, value } = evt.target;
    setFormData(oldFormData => ({
      ...oldFormData,
      [name]: value
    }))
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="height">Height:</label>
      <input
        type="number"
        id="height"
        name="height"
        value={formData.height}
        onChange={handleChange}
        style={{ margin: "10px" }}
      />
      <label htmlFor="width">Width:</label>
      <input
        type="number"
        id="width"
        name="width"
        value={formData.width}
        onChange={handleChange}
        style={{ margin: "10px" }}
      />
      <label htmlFor="color">Color:</label>
      <input
        id="color"
        name="color"
        value={formData.color}
        onChange={handleChange}
        style={{ margin: "10px" }}
      />
      <button type="submit">Submit</button>
    </form>
  )
}

export default BoxForm;