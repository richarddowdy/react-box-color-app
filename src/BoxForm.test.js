import React from "react";
import { render } from "@testing-library/react";
import BoxForm from "./BoxForm";

it("renders without crashing", function () {
  render(<BoxForm />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<BoxForm/>);
  expect(asFragment()).toMatchSnapshot();
});