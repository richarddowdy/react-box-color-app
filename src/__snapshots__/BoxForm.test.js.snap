// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      name="height"
      style="margin: 10px;"
      type="number"
      value="0"
    />
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      name="width"
      style="margin: 10px;"
      type="number"
      value="0"
    />
    <label
      for="color"
    >
      Color:
    </label>
    <input
      id="color"
      name="color"
      style="margin: 10px;"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
